presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java

```diff
@@ -134,6 +134,7 @@ public void configure(Binder binder)
         binder.bind(PrestoS3ClientFactory.class).in(Scopes.SINGLETON);
 
         binder.bind(DirectoryLister.class).annotatedWith(ForCachingDirectoryLister.class).to(HadoopDirectoryLister.class).in(Scopes.SINGLETON);
+        binder.bind(CachingDirectoryLister.class).in(Scopes.SINGLETON);
```

Comment:
Why is this needed?

Reply:
To Inject CachingDirectoryLister in DirectoryListCacheInvalidationProcedure, Explicit binding is required for this

Reply:
Could we inject a `DirectoryLister` and do a check if it's an instance of `CachingDirectoryLister`?

Reply:
Yes, I think we could do that